File: php/src/GildedRoseItem/SulfurasItem.php
<?php

declare(strict_types=1);

namespace GildedRose\GildedRoseItem;

use GildedRose\GildedRoseItem;
use GildedRose\Item;

class SulfurasItem implements GildedRoseItem
{
    public const NAME = 'Sulfuras, Hand of Ragnaros';

    public function __construct(
        private Item $item
    ) {
    }

    public function ageByOneDay(): void
    {
    }
}

File: Java/src/main/java/com/gildedrose/SulfurasItem.java
package com.gildedrose;

public class SulfurasItem extends UpdatableItem {
    public SulfurasItem(int sellIn, int quality) {
        super("Sulfuras, Hand of Ragnaros", sellIn, quality);
    }
    @Override
    public void update() {
        // Legendary item, does not change
    }
} 